invenio_app_ils/records_relations/retriever.py

```diff
@@ -170,6 +170,11 @@ def get(self):
                 r = self._build_relation_obj(sibling_pid, name)
                 relations.setdefault(name, [])
                 relations[name].append(r)
+
+            # Pre-requisite: fields being sorted should be of the same type.
+            if sort_by and name in sort_by and name in relations.keys():
+                relations[name].sort(key=lambda rec: rec["record_metadata"][name])
```

Comment:
This is not good enough: we are sorting too late, sorting the results of what was retrieved from the db with a default sorting.

If this is what we want to do, then it is better to have the client, consuming the REST API response, to sort the results.
If instead we want to drive the sorting from the backend, then the sorting param needs to be applied to the db query, probably [here](https://github.com/inveniosoftware/invenio-app-ils/blob/master/invenio_app_ils/relations/api.py#L116) and in all other methods that fetch the relations from the db.
However this is probably very tricky, let's discuss.

Reply:
Discussed IRL, decision to be taken.